### PR TITLE
python: Notes for scrabble score

### PR DESCRIPTION
**Pull request by jeffdparker**
I propose a focus on comprehensions and inverting dictionary as the key points to the exercise.

### CODE REVIEW

**Comment by jeffdparker** [comment]
OK.  Derk-Jan has assigned this to me.  What is my next step? I see two choices: 
Close and comment
Comment

I am selecting comment as the most harmless

File: tracks/python/exercises/scrabble-score/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+K                                  5
+J, X                               8
+Q, Z                               10
+```
+
+It is a useful exercise to use this table to build a dictionary that 
+maps numbers to strings, and then invert the dictionary
+to get a mapping from letters to their value in the game.
+
+This is less prone to error, and the editing is straigforward,
```
Comment (cmccandless) [inline tracks/python/exercises/scrabble-score/mentoring.md]:
```suggestion
This is less prone to error, and the editing is straightforward, 
```

File: tracks/python/exercises/scrabble-score/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    10: 'Q, Z'
+    }
+
+def rev(d: dict) -> dict:
+    return { ch:val for val in d for ch in d[val] if ch.isalpha() }
+
+def score(word: str) -> int:
+    letter_value = rev(numbers_to_letters)
+    return sum( [ letter_value[ch] for ch in word.upper() ] )
+```
```
Comment (cmccandless) [inline tracks/python/exercises/scrabble-score/mentoring.md]: If I understand this snippet correctly, the each value of `numbers_to_letters` should either be a list of single-character strings, or a single string with no punctuation or spacing.

File: tracks/python/exercises/scrabble-score/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+```python
+numbers_to_letters = {
+    1: 'A, E, I, O, U, L, N, R, S, T',
+    2: 'D, G',
+    3: 'B, C, M, P',
+    4: 'F, H, V, W, Y',
+    5: 'K',
+    8: 'J, X',
+    10: 'Q, Z'
+    }
```
Comment (cmccandless) [inline tracks/python/exercises/scrabble-score/mentoring.md]: See above.

File: tracks/python/exercises/scrabble-score/mentoring.md (hunk 2 above)
Comment (jeffdparker) [reply tracks/python/exercises/scrabble-score/mentoring.md]: No, the point of all this is that we can take the problem statement as it is, and paste it into the solution, with some minor work to remove non-letters.  

I am asking students to do "extra" work - the payoff is that they have much less work to do in grooming the definition of the dictionary.

File: tracks/python/exercises/scrabble-score/mentoring.md (hunk 3 above)
Comment (jeffdparker) [reply tracks/python/exercises/scrabble-score/mentoring.md]: See above.

**Comment by jeffdparker** [comment]
I would like to decline these edits - what is the right way to do so?

File: tracks/python/exercises/scrabble-score/mentoring.md (hunk 1 above)
Comment (jeffdparker) [reply tracks/python/exercises/scrabble-score/mentoring.md]: Thanks!

**Comment by cmccandless** [comment]
Typically, it is best to let the reviewer resolve (either after reviewing new changes or reviewing defense against requested changes). I have marked both as resolved and approved.

@SleeplessByte , please merge if there are no other issues.

**Comment by SleeplessByte** [comment]
@jeffdparker thank you 🎉

**Pull request merged**